### PR TITLE
Added known issue entry for synthetic source bug.

### PR DESCRIPTION
**Pull request by martijnvg**
The bug was fixed as part of #112713

### CODE REVIEW

File: docs/reference/release-notes/8.15.0.asciidoc [hunk 1]
```diff
@@ -44,6 +44,15 @@ To work around this issue, you have a number of options:
 <<esql-kibana-enable,disable ES|QL queries in {kib}>>
 ** Change the default data view in Discover to a smaller set of indices and/or one with fewer mapping conflicts.
 
+* Synthetic source bug. Synthetic source may fail generating the _source at runtime, causing failures in get APIs or
+partial failures in the search APIs. The result is that for the affected documents the _source can't be retrieved.
+There is no workaround and the only option to is to upgrade to 8.15.2 when released.
++
+If you use synthetic source then you may be affected by this bug if the following it true:
```
Comment (lkts) [inline docs/reference/release-notes/8.15.0.asciidoc:51]:
```suggestion
If you use synthetic source then you may be affected by this bug if the following is true:
```

File: docs/reference/release-notes/8.15.0.asciidoc [hunk 2]
```diff
@@ -44,6 +44,15 @@ To work around this issue, you have a number of options:
...
+** If you have more fields then the `index.mapping.total_fields.limit` setting allows.
+** If you use dynamic mappings and the `index.mapping.total_fields.ignore_dynamic_beyond_limit` setting is enabled.
+** If you use `nested` field types in your mappings.
```
Comment (lkts) [inline docs/reference/release-notes/8.15.0.asciidoc:54]: I am not sure why `nested` is mentioned. AFAIK it's not related?

File: docs/reference/release-notes/8.15.1.asciidoc [hunk 3]
```diff
@@ -24,13 +24,21 @@ To work around this issue, you have a number of options:
...
 ** If the affected index already contains documents with a dense vector field, force merge the index to a single segment.
 ** If the affected index does not already contain documents with a dense vector field, index a document with a dense vector field
 and then force merge to a single segment.
 ** If the affected index's `dense_vector` fields are unused, reindex without the `dense_vector` fields.
 
+* Synthetic source bug. Synthetic source may fail generating the _source at runtime, causing failures in get APIs or
+partial failures in the search APIs. The result is that for the affected documents the _source can't be retrieved.
+There is no workaround and the only option to is to upgrade to 8.15.2 when released.
++
+If you use synthetic source then you may be affected by this bug if the following it true:
```
Comment (lkts) [inline docs/reference/release-notes/8.15.1.asciidoc:38]:
```suggestion
If you use synthetic source then you may be affected by this bug if the following is true:
```

File: docs/reference/release-notes/8.15.0.asciidoc (hunk 2 above)
Comment (martijnvg) [reply docs/reference/release-notes/8.15.0.asciidoc]: Yes, it is removed. Nested or arrays have nothing to doing with this. It is about when `index.mapping.total_fields.limit` is exceeded or `index.mapping.total_fields.ignore_dynamic_beyond_limit` is configured.

**Review by lkts — APPROVED** [review]

**Pull request merged**